tests/instruments/test_ship_underwater_st.py

```diff
@@ -1,29 +1,102 @@
+
+    # variabes we are going to compare between expected and actual observations
+    variables = ["salinity", "temperature", "lat", "lon"]
+
+    # where to sample
+    sample_points = [
+        Spacetime(Location(3, 0), base_time + datetime.timedelta(seconds=0)),
+        Spacetime(Location(7, 0), base_time + datetime.timedelta(seconds=1)),
+    ]
+
+    # expected observations at sample points
+    expected_obs = [
+        {
+            "salinity": 1,
+            "temperature": 2,
+            "lat": 3,
+            "lon": 0,
+            "time": base_time + datetime.timedelta(seconds=0),
+        },
+        {
+            "salinity": 5,
+            "temperature": 6,
+            "lat": 7,
+            "lon": 0,
+            "time": base_time + datetime.timedelta(seconds=1),
+        },
+    ]
+
+    # create fieldset based on the expected observations
     fieldset = FieldSet.from_data(
-        {"U": 0, "V": 0, "S": 0, "T": 0},
+        {
+            "U": np.zeros((2, 2)),
+            "V": np.zeros((2, 2)),
+            "salinity": [
+                [expected_obs[0]["salinity"], 0],
+                [0, expected_obs[1]["salinity"]],
+            ],
+            "temperature": [
+                [expected_obs[0]["temperature"], 0],
+                [0, expected_obs[1]["temperature"]],
+            ],
+        },
         {
             "lon": 0,
-            "lat": 0,
-            "time": [np.datetime64("1950-01-01") + np.timedelta64(632160, "h")],
+            "lat": np.array([expected_obs[0]["lat"], expected_obs[1]["lat"]]),
+            "time": np.array(
+                [
+                    np.datetime64(expected_obs[0]["time"]),
+                    np.datetime64(expected_obs[1]["time"]),
+                ]
+            ),
         },
     )
 
-    sample_points = [Spacetime(Location(0, 0), 0)]
+    out_file_name = tmp_dir_factory(suffix=".zarr")
 
     simulate_ship_underwater_st(
         fieldset=fieldset,
-        out_file_name="test",
+        out_file_name=out_file_name,
         depth=DEPTH,
         sample_points=sample_points,
     )
+
+    results = xr.open_zarr(out_file_name)
+
+    # below we assert if output makes sense
+    assert len(results.trajectory) == 1  # expect a singe trajectory
+    traj = results.trajectory.item()
+    assert len(results.sel(trajectory=traj).obs) == len(
+        sample_points
+    )  # expect as many obs as sample points
+
+    # for every obs, check if the variables match the expected observations
+    for i, (obs_i, exp) in enumerate(
+        zip(results.sel(trajectory=traj).obs, expected_obs, strict=True)
+    ):
+        obs = results.sel(trajectory=traj, obs=obs_i)
+        for var in variables:
+            obs_value = obs[var].values.item()
+            exp_value = exp[var]
+            assert np.isclose(
+                obs_value, exp_value
+            ), f"Observation incorrect {i=} {var=} {obs_value=} {exp_value=}."
```

Comment:
Would it make more sense to print `obs_i`?
```suggestion
            ), f"Observation incorrect {obs_i=} {var=} {obs_value=} {exp_value=}."
```

Reply:
correct, adjusted and enumerate removed